src/Checkout/Order/Shipping/ShippingSubscriber.php

```diff
@@ -73,7 +73,7 @@ public function onOrderDeliveryWritten(EntityWrittenEvent $event): void
             }
 
             $changeSet = $writeResult->getChangeSet();
-            if ($changeSet && !$changeSet->hasChanged('tracking_codes')) {
+            if ((!$changeSet && !isset($writeResult->getPayload()['trackingCodes'])) || ($changeSet && !$changeSet->hasChanged('tracking_codes'))) {
                 continue;
             }
```

Comment:
to make it more readable, would you mind splitting this into two `if` conditions with `continue;`? Maybe even with a comment as to why we can skip without a ChangeSet 😉